uaclient/files/files.py

```diff
@@ -64,6 +64,36 @@ def delete(self):
         system.ensure_file_absent(self.path)
 
 
+class ProJSONFile:
+    def __init__(
+        self,
+        name: str,
+        directory: str = defaults.DEFAULT_DATA_DIR,
+        private: bool = True,
+    ):
+        self._file = UAFile(name=name, directory=directory, private=private)
+
+    def write(self, content: Dict[str, Any]):
+        self._file.write(
+            content=json.dumps(content, cls=util.DatetimeAwareJSONEncoder)
+        )
+
+    def read(self) -> Optional[Dict[str, Any]]:
+        content = self._file.read()
+
+        if content:
+            return json.loads(content, cls=util.DatetimeAwareJSONDecoder)
+
+        return None
+
+    def delete(self):
+        return self._file.delete()
+
+    @property
+    def is_present(self):
+        return self._file.is_present
+
```

Comment:
Was there a specific reason for introducing this layer of abstraction other than the json serialization/de serialization ? It would be helpful to understand the motivation behind this change (which i think is a great addition).

Also, do we want to handle any errors when doing when reading and writing through json?